src/components/sidebar/sidebar.component.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Icon from '../icon/icon';
+import Modal from '../modal/modal';
+import SidebarStyle from './sidebar.style';
+import './sidebar.scss';
+
+class Sidebar extends Modal {
+  /** Returns classes for the component. */
+  get mainClasses() {
+    return classNames(
+      'carbon-sidebar',
+      this.props.className
+    );
+  }
+
+  /** Returns the markup for the close icon. */
+  get closeButton() {
+    if (this.props.onCancel) {
+      return (
+        <span className='carbon-sidebar__close'>
+          <Icon
+            className='carbon-sidebar__close-icon'
```

Comment:
please rewrite these classes to styled-components

Reply:
Done :)